ckanext/scheming/tests/test_validation.py

```diff
@@ -287,10 +281,249 @@ def test_datetime_field_valid_separate_time(self):
+            a_relevant_datetime_tz='2014-01-01T12:35:00-05:00',
+        )
+        assert_equals(d['a_relevant_datetime_tz'], '2014-01-01T17:35:00')
+
+    def test_date_field_valid_date_datetime(self):
+        lc = LocalCKAN()
+        d = lc.action.package_create(
+            type='camel-photos',
+            name='fred_datetime_tz10',
+            a_relevant_datetime_tz=datetime.datetime(2014, 1, 1, 12, 35),
+        )
+        assert_equals(d['a_relevant_datetime_tz'], '2014-01-01T12:35:00')
+
+    def test_date_field_datetime_convert_to_utc(self):
+        lc = LocalCKAN()
+        d = lc.action.package_create(
+            type='camel-photos',
+            name='fred_datetime_tz11',
+            a_relevant_datetime_tz=datetime.datetime(
+                2014, 1, 1, 12, 35, tzinfo=pytz.timezone('America/New_York')
+            ),
+        )
+
+    def test_datetime_field_rejects_invalid_separate_date(self):
+        lc = LocalCKAN()
+        try:
+            lc.action.package_create(
+                type='camel-photos',
+                name='fred_datetime_tz12',
+                a_relevant_datetime_tz_date='31/11/2014',
+            )
+        except ValidationError as e:
+            assert_equals(e.error_dict['a_relevant_datetime_tz_date'],
+                          ['Date format incorrect'])
+        else:
+            raise AssertionError('ValidationError not raised')
+
+    def test_datetime_field_rejects_invalid_separate_time(self):
+        lc = LocalCKAN()
+        try:
+            lc.action.package_create(
+                type='camel-photos',
+                name='fred_datetime_tz13',
+                a_relevant_datetime_tz_date='2014-01-01',
+                a_relevant_datetime_tz_time='12:35:aa',
+            )
+        except ValidationError as e:
+            assert_equals(e.error_dict['a_relevant_datetime_tz_time'],
+                          ['Time format incorrect'])
+        else:
+            raise AssertionError('ValidationError not raised')
+
+    def test_datetime_field_rejects_invalid_separate_tz(self):
+        lc = LocalCKAN()
+        try:
+            lc.action.package_create(
+                type='camel-photos',
+                name='fred_datetime_tz14',
+                a_relevant_datetime_tz_date='2014-01-01',
+                a_relevant_datetime_tz_time='12:35:00',
+                a_relevant_datetime_tz_tz='Krypton/Argo City',
+            )
+        except ValidationError as e:
+            assert_equals(e.error_dict['a_relevant_datetime_tz_tz'],
+                          ['Invalid timezone'])
+        else:
+            raise AssertionError('ValidationError not raised')
+
+    def test_datetime_field_rejects_time_only(self):
+        lc = LocalCKAN()
+        try:
+            lc.action.package_create(
+                type='camel-photos',
+                name='fred_datetime_tz15',
+                a_relevant_datetime_tz_date='',
+                a_relevant_datetime_tz_time='12:35:aa',
```

Comment:
this should be a valid time and no date passed at all (no empty string passed), right?